pk3DS/Subforms/Gen7/GameSummary7.cs

```diff
@@ -0,0 +1,128 @@
+using pk3DS.Core.Structures;
+using pk3DS.Core.Structures.PersonalInfo;
+
+namespace pk3DS
+{
+    public partial class GameSummary7 : Form
+    {
+        public GameSummary7()
+        {
+            InitializeComponent();
+            Setup();
+        }
+        #region Global Variables
+
+        private readonly List<PersonalInfo> pkms = Main.Config.Personal.Table.ToList();
+        private readonly List<Move> moves = Main.Config.Moves.ToList();
+        private readonly List<int> types = Main.Config.Personal.Table.SelectMany(pkm => pkm.Types).ToHashSet().ToList();
+        private readonly string[] typeNames = Main.Config.getText(TextName.Types);
+        
+        private readonly List<int> tms = TMEditor7.getTMHMList().Select(tm => Convert.ToInt32(tm)).ToList();
+        private readonly int[][] tutors = TutorEditor7.getTutorList();
+       
+        private Dictionary<int, double> moveTypeCounts = new Dictionary<int, double>();
+        private Dictionary<int, double> pokemonMatchingTypeCounts = new Dictionary<int, double>();
+        private Dictionary<int, double> pokemonNonMatchingTypeCounts = new Dictionary<int, double>();
+        private Dictionary<int, double> matchingTypeCounts = new Dictionary<int, double>();
+        private Dictionary<int, double> nonMatchingTypeCounts = new Dictionary<int, double>();
+        
+        private Dictionary<int, double> tutorMoveTypeCounts = new Dictionary<int, double>();
+        private Dictionary<int, double> tutorPokemonMatchingTypeCounts = new Dictionary<int, double>();
+        private Dictionary<int, double> tutorPokemonNonMatchingTypeCounts = new Dictionary<int, double>();
+        private Dictionary<int, double> tutorMatchingTypeCounts = new Dictionary<int, double>();
+        private Dictionary<int, double> tutorNonMatchingTypeCounts = new Dictionary<int, double>();
+        
+        #endregion
+        
+        private void Setup()
+        {
+            foreach (int type in types)
+            {
+                moveTypeCounts[type] = 0;
+                pokemonMatchingTypeCounts[type] = 0;
+                pokemonNonMatchingTypeCounts[type] = 0;
+                matchingTypeCounts[type] = 0;
+                nonMatchingTypeCounts[type] = 0;
+
+                tutorMoveTypeCounts[type] = 0;
+                tutorPokemonMatchingTypeCounts[type] = 0;
+                tutorPokemonNonMatchingTypeCounts[type] = 0;
+                tutorMatchingTypeCounts[type] = 0;
+                tutorNonMatchingTypeCounts[type] = 0;
+            }
+
+            foreach (PersonalInfo pkm in Main.Config.Personal.Table)
+            {
+                for (int tmIndex = 0; tmIndex < tms.Count; tmIndex++)
+                {
+                    Move m = moves[tms[tmIndex]];
+                    int t = m.Type;
+                    moveTypeCounts[t] = moveTypeCounts[t] + 1;
+                    int pkmHasTm = pkm.TMHM[tmIndex] ? 1 : 0;
+                    bool isSTAB = pkm.Types.ToList().Contains(t);
```

Comment:
Don't keep turning pkm.Types into a list. One way to avoid this is to save pkm.Types.ToList as a variable just before the current for loop.